Jack/Property.fs

```diff
@@ -43,6 +43,13 @@ module Property =
         else
             Failure [] |> ofResult
 
+    let private convert (value : 'T) =
+        match box value with
+        | :? Lazy<bool> as b -> ofBool b.Value
+        | :? Property   as p -> p
+        | :? bool       as b -> ofBool b
+        | _                  -> ofBool true
```

Comment:
I suspect that as always, the main hurdle is not the engineering work, but convincing the F# team at Microsoft that's it's a good fit. Apparently the best thing we can do to help is to get lots of people to say that they really want this feature.